spa/src/components/modals/AvatarModal.vue

```diff
@@ -1,24 +1,43 @@
 <template>
   <Modal>
     <template v-slot:header>
-        <button type="button" class="btn-ui-inline" @click="close('Modal closed')">X</button>
-        <button type="button" class="btn-ui-inline" @click="openInfoModal">&lt;</button>
+      <button type="button"
+              class="btn-ui-inline"
+              @click="close('Modal closed')">X</button>
+      <button type="button"
+              class="btn-ui-inline"
+              @click="openInfoModal">&lt;</button>
     </template>
+    <!-- avatars/avlib.html -->
     <template v-slot:body>
       <div class="flex-1">
         <h3 align="center">Update Your Avatar</h3>
-        <p align="center" v-if="showError" class="text-red-500">{{ error }}</p>
-        <p align="center" v-if="showSuccess" color="#00FF00">
+        <p align="center"
+           v-if="showError"
+           class="text-red-500">{{ error }}</p>
+        <p align="center"
+           v-if="showSuccess"
+           class="text-green"
+           >
           Account Details Updated!
         </p>
         <div align="center">
-          <select v-model="avatarId">
-            <option v-for="avatar in avatars" :value="avatar.id" :key="avatar.id">
-              {{ avatar.name }}
-            </option>
-          </select>
+
+          <button class="btn-ui"
+                  @click="openAvatarUploadModal">Upload Avatar</button>
+          <div class="grid lg:grid-cols-4 md:grid-cols-3 gap-6">
+            <div v-for="(avatar, key) in avatars"
+                 :key="key">
+              <img :src="'/assets/avatars/' + avatar.directory + '/' + avatar.image"
+                   style="max-width:100%;max-height:250px;height:auto;width:auto;" />
+              <div>{{ avatar.name }}</div>
+              <div class="text-green" v-if="avatar.id == avatarId">Current Avatar</div>
+              <button v-else type="button"
+                      class="btn-ui"
+                      @click="save(avatar.id)">Use Avatar</button>
```

Comment:
Got this error message on trying to pick a private avatar:<br>
<img width="1012" alt="Screenshot 2024-05-02 at 10 37 15 PM" src="https://github.com/CybertownRevival/ctr/assets/98184756/c268ac1e-4f8c-429f-928e-d5eeaad49996">